File: python/2.7/observer/com/weather/observer/StatisticsDisplay.py
from Observer import *
from DisplayElement import *

class StatisticsDisplay(Observer, DisplayElement):
    def __init__(self, weatherData):
        self.maxTemp = 0.0
        self.minTemp = 200
        self.tempSum = 0.0
        self.numReadings = 0
        self.weatherData = weatherData
        self.weatherData.registerObserver(self)

    def update(self, temp, humidity, pressure):
        self.tempSum += temp
        self.numReadings += 1

        if(temp > self.maxTemp):
            self.maxTemp = temp
        if(temp < self.minTemp):
            self.minTemp = temp

        self.display()

    def display(self):
        print "Avg/Max/Min temperature = %s/%s/%s"%(self.tempSum/self.numReadings, self.maxTemp, self.minTemp)

File: js/node/observer/com/weather/observer/StatisticsDisplay.js
require('./Observer');
require('./DisplayElement');

StatisticsDisplay = function(weatherData) {
    Observer.apply(this);
    DisplayElement.apply(this);
    this.maxTemp = 0.0;
    this.minTemp = 200;
    this.tempSum = 0.0;
    this.numReadings = 0;
    this.weatherData = weatherData;
    this.weatherData.registerObserver(this);
    this.update = function(temp, humidity, pressure) {
        this.tempSum += temp;
        this.numReadings += 1;

        if(temp > this.maxTemp)
            this.maxTemp = temp;
        if(temp < this.minTemp)
            this.minTemp = temp;

        this.display();
    }
    this.display = function() {
        console.log("Avg/Max/Min temperature = " + this.tempSum/this.numReadings + "/" + this.maxTemp + "/" + this.minTemp);
    }
}